lib/processors/resourceListCreator.js

```diff
@@ -0,0 +1,646 @@
+	 * parameters: excludes can be denoted with a leading '-' or '!' and order is significant.
+	 * Later filters can override the result of earlier ones.
+	 *
+	 * If no component is given, orphans will only be reported but not added to any component (default).
+	 *
+	 * @since 1.29.1
+	 * /
+	@Parameter
+	private String externalResources;
+*/
+
+/**
+ * List of resource patterns that describe all debug resources.
+ *
+ * @since 1.29.1
+ */
+const DEFAULT_DEBUG_RESOURCES_FILTER = [
+	"**/*-dbg.js",
+	"**/*-dbg.controller.js",
+	"**/*-dbg.designtime.js",
+	"**/*-dbg.support.js",
+	"**/*-dbg.view.js",
+	"**/*-dbg.fragment.js",
+	"**/*-dbg.css",
+	"**/*.js.map"
+];
+
+/**
+ * List of resource patterns that describe bundled resources.
+ *
+ * @since 1.29.1
+ */
+const DEFAULT_BUNDLE_RESOURCES_FILTER = [
+	"**/Component-preload.js",
+	"**/library-preload.js",
+	"**/library-preload-dbg.js",
+	"**/library-preload.json",
+	"**/library-h2-preload.js",
+	"**/designtime/library-preload.designtime.js",
+	"**/library-preload.support.js",
+	"**/library-all.js",
+	"**/library-all-dbg.js"
+];
+
+/**
+ * List of resource patterns that describe all designtime resources.
+ *
+ * @since 1.31.0
+ */
+const DEFAULT_DESIGNTIME_RESOURCES_FILTER = [
+	"**/designtime/*",
+	"**/*.designtime.js",
+	"**/*.control",
+	"**/*.interface",
+	"**/*.type",
+	"**/themes/*/*.less",
+	"**/library.templates.xml",
+	"**/library.dependencies.xml",
+	"**/library.dependencies.json"
+];
+
+/**
+ * List of resource patterns that describe all support (assistant) resources.
+ *
+ * @since 1.53.0
+ */
+const DEFAULT_SUPPORT_RESOURCES_FILTER = [
+	"**/*.support.js"
+];
+
+const LOCALE = /^((?:[^/]+\/)*[^/]+?)_([A-Z]{2}(?:_[A-Z]{2}(?:_[A-Z0-9_]+)?)?)(\.properties|\.hdbtextbundle)$/i;
+const THEME = /^((?:[^/]+\/)*)themes\/([^/]+)\//;
+
+class ResourceCollector {
+	constructor(pool, filter) {
+		this._pool = pool;
+		this._filter = filter != null ? filter : new ResourceFilterList();
+		this._resources = new Map();
+		this._components = new Map();
+		this._themePackages = new Map();
+	}
+
+	setExternalResources(list) {
+		this._externalResources = list;
+	}
+
+	visitResource(virPath) {
+		if ( !virPath.startsWith("/resources/") ) {
+			log.warn(`non-runtime resource ${virPath} ignored`);
+			return;
+		}
+		const name = virPath.slice("/resources/".length);
+		if ( this._filter.matches(name) ) {
+			this._resources.set(name, new ResourceInfo(name));
+
+			const p = name.lastIndexOf("/");
+			const prefix = name.substring(0, p + 1);
+			const basename = name.substring(p + 1);
+			if ( basename.match("[^/]*\\.library|Component\\.js|manifest\\.json") && !this._components.has(prefix) ) {
+				this._components.set(prefix, new ResourcesList(prefix));
+			}
+			// a .theme file within a theme folder indicates a library/theme package
+			// Note: ignores .theme files in library folders
+			if ( name.match("(?:[^/]+/)*themes/[^/]+/(?:\\.theming|library.source.less)") && !this._themePackages.has(prefix) ) {
```

Comment:
In the Maven tooling, the `library.source.less` was not used as a marker, but the ui5-tooling does not create the expected `.theming` files. @matz3 What is the right approach here?  Should we create the `.theming` files or chose a different approach?

Reply:
We need to create the files in order to support the Theming Engine. This becomes relevant when our main delivery is built with the UI5 Tooling.

For now the `library.source.less` as an indicator is IMO fine. We also use it within the theme build as the files are required for the build (which was not the case in the legacy Maven build).

Reply:
We only do so in the internal theme build (which produces the current delivery for SAPUI5 / OpenUI5 (except the npm packages).
So I think we need to stay with `library.source.less` for the time being.